arch/risc-v/src/mpfs/mpfs_plic.c

```diff
@@ -0,0 +1,131 @@
+/****************************************************************************
+ * arch/risc-v/src/mpfs/mpfs_plic.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdint.h>
+#include <stdio.h>
+#include <assert.h>
+#include <debug.h>
+
+#include <nuttx/arch.h>
+#include <arch/irq.h>
+
+#include "mpfs.h"
+#include "mpfs_plic.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: mpfs_plic_get_iebase
+ *
+ * Description:
+ *   Context aware way to query PLIC interrupt enable base address
+ *
+ * Returned Value:
+ *   Interrupt enable base address
+ *
+ ****************************************************************************/
+
+uintptr_t mpfs_plic_get_iebase(void)
+{
+  uint32_t *iebase;
```

Comment:
```suggestion
  uintptr_t iebase;
```

Reply:
Sure, I just copy-pasted the existing code from mpfs_irq.c but of course I can also do styling